apex/contrib/optimizers/distributed_fused_adam.py

```diff
@@ -2329,7 +2392,8 @@ def step(
                     )
 
         # Apply optimizer step
-        self.state["step"] += 1
+        self.state["step"] += 1 if not self.capturable else \
+            (self._dummy_overflow_buf != 1).to(torch.int)
```

Comment:
Q: where would we decrement this value when `self.capturable` is `True` and invalid grads are found?

Reply:
As you know `self.state["step"]` is to track how many steps the optimizer has advanced, it is used for bias correction in the CUDA kernel. When invalid grads are found, `self._dummy_overflow_buf` is 1, then it's `self.state["step"] += 0`, otherwise it's `self.state["step"] += 1`.

Reply:
uh, I misread it, thank you for correcting me.

another question: would this be really host-device sync free?

Reply:
Yes, in this case distributed fused adam is sync-free.